UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -0,0 +1,370 @@
+
+	public List<ImplantBase> ImplantList => implantList;
+
+	public event Action<GameObject> applyDamageEvent;
+
+	public event Action OnDeathNotifyEvent;
+
+	//how on fire we are, sames as tg fire_stacks. 0 = not on fire.
+	//It's called "stacks" but it's really just a floating point value that
+	//can go up or down based on possible sources of being on fire. Max seems to be 20 in tg.
+	[SyncVar(hook = nameof(SyncFireStacks))]
+	private float fireStacks;
+	private float maxFireStacks = 5f;
+	private bool maxFireStacksReached = false;
+
+	/// <summary>
+	/// How on fire we are. Exists client side - synced with server.
+	/// </summary>
+	public float FireStacks => fireStacks;
+
+	/// <summary>
+	/// Client side event which fires when this object's fire status changes
+	/// (becoming on fire, extinguishing, etc...). Use this to update
+	/// burning sprites.
+	/// </summary>
+	[NonSerialized]
+	public FireStackEvent OnClientFireStacksChange = new FireStackEvent();
+
+	private ObjectBehaviour objectBehaviour;
+	public ObjectBehaviour OBehavior => objectBehaviour;
+
+
+	public virtual void Awake()
+	{
+		EnsureInit();
+	}
+
+	void OnEnable()
+	{
+		UpdateManager.Add(CallbackType.UPDATE, UpdateMe);
+		UpdateManager.Add(PeriodicUpdate, 1f);
+	}
+
+	void OnDisable()
+	{
+		UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+		UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, PeriodicUpdate);
+	}
+
+	private void EnsureInit()
+	{
+		if (registerTile) return;
+		registerTile = GetComponent<RegisterTile>();
+		respiratorySystem = GetComponent<RespiratorySystemBase>();
+		circulatorySystem = GetComponent<CirculatorySystemBase>();
+		objectBehaviour = GetComponent<ObjectBehaviour>();
+		//Always include blood for living entities:
+	}
+
+	public override void OnStartServer()
+	{
+		EnsureInit();
+		mobID = PlayerManager.Instance.GetMobID();
+
+		//Generate BloodType and DNA
+		DNABloodType = new DNAandBloodType();
+	}
+
+	/// <summary>
+	/// Adds a new implant to the health master.
+	/// This is NOT how implants should be added, it is called automatically by the body part container system!
+	/// </summary>
+	/// <param name="implant"></param>
+	public void AddNewImplant(ImplantBase implant)
+	{
+		implantList.Add(implant);
+	}
+
+	public void RemoveImplant(ImplantBase implantBase)
+	{
+		implantList.Remove(implantBase);
+	}
+
+
+	public override void OnStartClient()
+	{
+		base.OnStartClient();
+		EnsureInit();
+		StartCoroutine(WaitForClientLoad());
+	}
+
+	private void UpdateMe()
+	{
+		foreach (ImplantBase implant in implantList)
+		{
+			implant.ImplantUpdate(this);
+		}
+	}
+
+	private void PeriodicUpdate()
+	{
+	}
+
+	IEnumerator WaitForClientLoad()
+	{
+		//wait for DNA:
+		while (string.IsNullOrEmpty(DNABloodTypeJSON))
+		{
+			yield return WaitFor.EndOfFrame;
+		}
+
+		yield return WaitFor.EndOfFrame;
+		DNASync(DNABloodTypeJSON, DNABloodTypeJSON);
+	}
+
+	// This is the DNA SyncVar hook
+	private void DNASync(string oldDNA, string updatedDNA)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldDNA'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6324510)

Reply:
Ignore this as i think its still needing for sync vars